### PR TITLE
Add `hugo list taxonomies` command

### PR DESCRIPTION
**Pull request by omajid**
This command operates in two ways:

- A simple invocation (`hugo list taxonomies`) prints out the taxonomies listed in config.toml

- An invocaiton with argumetns (`hugo list taxonomies tags`) prints out all the values of the that specific taxonomy used in the site. This makes it easy to find, for example, all the categories used in the site (`hugo list taxonomies categories`) or all the values of tags (`hugo list taxonomies tags`).

### CODE REVIEW

**Comment by omajid** [comment]
Here's how an emacs interface for editing hugo is re-implementing this: https://github.com/masasam/emacs-easy-hugo/blob/2dfdcdbcd8eb81c3fd631372d7dceb4f432d5f1e/easy-hugo.el#L1877-L1902

It manually searches (using regular expressions) across all "source" files to figure out the valid values of categories. The code is duplicated for a similar functionality for tags.

**Comment by omajid** [comment]
Anyone willing to review and merge this?

**Comment by omajid** [comment]
>  If this is a feature request, and you feel that it is still relevant and valuable, please tell us why.

It's adding a features that existing tools for Hugo need, as well as a general feature that makes the hugo cli nice to use.

File: go.mod [hunk 1]
```diff
@@ -39,6 +39,7 @@ require (
 	github.com/pelletier/go-toml v1.6.0 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/rogpeppe/go-internal v1.5.1
+	github.com/rogpeppe/godef v1.1.2 // indirect
```
Comment (moorereason) [inline go.mod:42]: I'm not sure about the `go.mod` and `go.sum` changes.  Are they relevant to this PR?

Comment (omajid) [reply go.mod:42]: No... when I built hugo, the build system made the changes to this file. Should I undo them?

Comment (moorereason) [reply go.mod:42]: I would undo the changes to the go.* files and then rebuild to see if they are added back.  If not, remove them from the PR.

**Pull request closed without merging**